File: src/20.ts
// Computes the sum of the digits of the factorial of a given number.
//
// See {@link https://projecteuler.net/problem=20}.
import _ from 'lodash';
import factorial from './core/factorial';

export default function compute(limit: number) {
  const text = _.toString(factorial(limit));

  let sum = 0;
  for (let i = 0; i < text.length; i++) {
    sum += _.toNumber(text[i]);
  }

  return sum;
}

File: project_euler/20.py
# Problem 20: Factorial Digit Sum
#
# Find the sum of the digits in the number 100!
#
# See https://projecteuler.net/problem=20
import math


def run(n=100) -> int:
    return sum(int(c) for c in str(math.factorial(n)))


def test_run():
    assert run() == 648
